### DISCUSSION TITLE
Expose paging response headers in AJAX requests

### DISCUSSION THREAD

**Discussion by MariaMozgunova** [openpotato/openplzapi.website#8] [Q&A] [answered]
I am making a jQuery AJAX request. I want to access paging response headers but they are not available on the response object.
It looks like the server should explicitly allow paging response headers so the client can read them.
Is that something that could be updated, please?
Reference: [Question on StackOverflow](https://stackoverflow.com/questions/37400091/missing-headers-in-response-for-jquery-request-with-cors)

**Comment by fstueber** [comment]
Sure, could you please give me a short jQuery AJAX request I can test against.

**Comment by fstueber** [comment]
Never mind, I know how to test this. Will try to fix this tomorrow.

**Comment by fstueber** [comment]
Quick test with JQuery 3.6 within the browser: The debugger shows me that the paging response headers are available. Could you please give me a short sample code to reproduce?

**Comment by MariaMozgunova** [comment]
Yes, the paging response headers are available in the browser. They are not available in the AJAX response though when I print all headers with `request.getAllResponseHeaders()`. The only headers the method outputs are: `content-type: application/json; charset=utf-8`.
The link in the original post suggests that to fix this, the server should send `Access-Control-Expose-Headers` with a list of paging headers so the client can access them.

**Comment by fstueber** [comment, marked as answer]
I have added `Access-Control-Expose-Headers`. Please check.

**Comment by MariaMozgunova** [comment]
Thank you, this works now
